ex_10.js

```diff
@@ -0,0 +1,10 @@
+const massId = [3, 1, 2]
+
+const summPrice = (products, massId) =>
+  products.reduce((accum, { id, price, currency }) => {
+    if (!accum[currency] && massId.includes(id))
+      return { ...accum, [currency]: price }
+    else if (accum[currency] && massId.includes(id))
```

Comment:
I recommend use  the second **if** instead of **else if**

accum[currency] && massId.includes(id) - you can put on variable and use hasCurrency
 if (!hasCurrency && hasProduct) {
 ...}
  if (hasCurrency && hasProduct) {
 ...}